### PR TITLE
Add tests for `assert_equal` and `assert_equal_constant`

### LINKED ISSUE #733: Add tests for `assert_equal` and `assert_equal_constant`
**Describe what you want implemented** 
Add in-circuit tests for the composer methods
- `assert_equal`
- `assert_equal_constant`

**Describe "Why" this is needed**
Increase test coverage

**Describe alternatives you've considered**
N/A

**Additional context**
Continuation of #731

### PR DESCRIPTION
**Pull request by moCello**
Resolves #733

### CODE REVIEW

File: tests/select_point.rs [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+            let w_result = composer.append_point(self.result);
+
+            let result_circuit =
+                composer.component_select_point(w_bit, w_point_a, w_point_b);
+
+            composer.assert_equal_point(w_result, result_circuit);
+
+            Ok(())
+        }
+    }
```
Comment (ureeves) [inline tests/select_point.rs:67]: Should we, for the sake of example completeness, constrait the `bit` to be 0 or 1?

File: tests/common/mod.rs [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+// Check that proof creation and verification of a satisfied circuit fails
+// when the public inputs from the test circuit does not match the ones from
+// the verifier circuit description
+// As this is a very specific test case, this function will not be used by all
+// tests.
+#[allow(dead_code)]
+pub(crate) fn check_satisfied_circuit_fails<C, R>(
+    prover: &Prover<C>,
+    verifier: &Verifier<C>,
+    pi_expected: &Vec<BlsScalar>,
```
Comment (ureeves) [inline tests/common/mod.rs:57]: Just in general taking a reference to a vector can be easily substituted by a slice to the items in the vector.

**Review by ureeves — APPROVED** [review]
One nit, and one circuit question, but not enough to block merging. LGTM

File: tests/select_point.rs (hunk 1 above)
Comment (moCello) [reply tests/select_point.rs:67]: You're completely right!

File: tests/common/mod.rs (hunk 2 above)
Comment (moCello) [reply tests/common/mod.rs:57]: For the tests I'll leave it as is if you don't mind

File: tests/select_point.rs (hunk 1 above)
Comment (moCello) [reply tests/select_point.rs:67]: On second thought though: I actually added tests that pass even though `bit` is not 1 or 0 to specifically test that behavior. So I might opt to add a comment, that clarifies that any `component_select` should _always_ constrain `bit` _before_ the component is called.

Comment (moCello) [reply tests/select_point.rs:67]: In any case: the `component_select*` tests were part of the last issue #732 so I'll go ahead and merge this one

**Pull request merged**